### PR TITLE
Fixing search_space initialization on JSMA

### LINKED ISSUE #841: JSMA only iterates once when clip_values is not set
I noticed some incomprehensible behavior in `art.attacks.evasion.SaliencyMapMethod()` (JSMA).
The method is supposed to work even without `clip_values` but I noticed that if it is not set, the method only perturbs 2 features and stops.
When I checked the code with more attention, I found that if `clip_values` is not set, the `search_space` (which is one of the stop conditions) is initialized with zeros. So right after the first iteration (which perturbs 2 features), it does not satisfy the condition.

What I think would be better is to initialize `search_space` with ones when `clip_values` are not set. When `clip_values` is set, `search_space` can be initialized with zeros and then the right features can be activated.

Actual implementation :
```
#Initialize the search space; optimize to remove features that can't be changed
search_space = np.zeros(batch.shape)
if self.estimator.clip_values is not None:
    clip_min, clip_max = self.estimator.clip_values
    if self.theta > 0:
        search_space[batch < clip_max] = 1
    else:
        search_space[batch > clip_min] = 1
```
Proposed solution :
```
#Initialize the search space; optimize to remove features that can't be changed
search_space = np.ones(batch.shape)
if self.estimator.clip_values is not None:
    search_space = np.zeros(batch.shape)
    clip_min, clip_max = self.estimator.clip_values
    if self.theta > 0:
        search_space[batch < clip_max] = 1
    else:
        search_space[batch > clip_min] = 1
```

### PR DESCRIPTION
**Pull request by mamerzouk**
# Description
   ####
Changing the initialization of `search_space` in JSMA to prevent it from stopping after only one iteration when `clip_value` is not set.

Fixes #841 

## Type of change

Please check all relevant options.

- [ ] Improvement (non-breaking)
- [x] Bug fix (non-breaking)
- [ ] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Testing

Please describe the tests that you ran to verify your changes. Consider listing any relevant details of your test configuration.

- [x] Generating adversarial examples on the intrusion detection dataset NSL-KDD.
- [x] Generating adversarial examples on the intrusion detection dataset CIDDS-001.

**Test Configuration**:
- Fedora 31 (Linux 5.3.7-301.fc31.x86_64)
- Python version 3.7.9
- ART dev_1.6.0
- PyTorch version

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: art/attacks/evasion/saliency_map.py [hunk 1]
```diff
@@ -109,8 +109,9 @@ def generate(self, x: np.ndarray, y: Optional[np.ndarray] = None, **kwargs) -> n
 
             # Main algorithm for each batch
             # Initialize the search space; optimize to remove features that can't be changed
-            search_space = np.zeros(batch.shape)
+            search_space = np.ones(batch.shape)
```
Comment (beat-buesser) [inline art/attacks/evasion/saliency_map.py:112]: Could you please move this line into an `else` block?

**Review by beat-buesser — CHANGES_REQUESTED** [review]
Hi @mamerzouk Thank you very much for your pull request fixing the search space issue you have reported. I have one small change request.

File: art/attacks/evasion/saliency_map.py (hunk 1 above)
Comment (mamerzouk) [reply art/attacks/evasion/saliency_map.py:112]: Sure ! Thank you for the review.

**Review by beat-buesser — APPROVED** [review]
Hi @mamerzouk Thank you very much for improving the Saliency Map attack!

**Pull request merged**